src/x/pool/object.go

```diff
@@ -172,3 +176,11 @@ func (p *objectPool) tryFill() {
 		}
 	}()
 }
+
+// Close shuts down background routines
+// TODO: if we ever need to shut down gracefully (or start using goleak in tests),
```

Comment:
you're going to be :( to hear, but lots of test already use leaktest, so this is going to cause issues.